internal/webhooks/cluster.go

```diff
@@ -256,19 +272,31 @@ func (webhook *Cluster) validateTopology(ctx context.Context, old, new *clusterv
 
 		// If the ClusterClass referenced in the Topology has changed compatibility checks are needed.
 		if old.Spec.Topology.Class != new.Spec.Topology.Class {
-			oldClusterClass := &clusterv1.ClusterClass{}
-
 			// Check to see if the ClusterClass referenced in the old version of the Cluster exists.
-			if err := webhook.Client.Get(ctx, client.ObjectKey{Namespace: old.Namespace, Name: old.Spec.Topology.Class}, oldClusterClass); err != nil {
+			oldClusterClass, err := webhook.getClusterClassForCluster(ctx, old)
+			if err != nil {
 				allErrs = append(
 					allErrs, field.Forbidden(
 						field.NewPath("spec", "topology", "class"),
-						fmt.Sprintf("old ClusterClass %v could not be found, changes to ClusterClass %v can not be validated for cluster %v",
-							old.Spec.Topology.Class, new.Spec.Topology.Class, new.Name)))
+						fmt.Sprintf("ClusterClass %v could not be found, changes to ClusterClass %v can not be validated for cluster %v",
```

Comment:
nit, The message might be misleading given that the Cluster is now using another ClusterClass and the ClusterClass itself is not changed.

Maybe:
```go
fmt.Sprintf("ClusterClass %v could not be found, changes between ClusterClass %v and %v can not be validated for Cluster %v",
```
(second parameter is old class, third is new)

Reply:
Another nit: I would use `%s` for strings instead of `%v`